src/Receiver.h

```diff
@@ -426,11 +413,17 @@ class Receiver {
         {}
 
         /**
-         * Peer destructor.
+         * Peer destructor. Only invoked from the destructor of Receiver.
          */
         ~Peer()
         {
-            scheduledMessages.clear();
+            // By the time we need to destruct a peer, all Message's coming from
+            // it should have been released.
+            assert(scheduledMessages.empty());
+
+            // To keep Peer (constructor) simple, we don't store a reference to
```

Comment:
Don't understand how this keeps the "constructor" simple? Did you mean "destructor"?

Reply:
Sorry for the confusion; I meant to say that we would like to provide a default constructor for `Peer` so it can be used with `std::unordered_map` easily. I have updated the documentation to be more clear.